plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -32,10 +33,21 @@ def __init__(
             name: Name of the task
             query: DuckDB query to execute
             inputs: The query parameters to be used while executing the query
+            md_secret: Secret containing a MotherDuck authentication token
         """
+        if md_secret is None:
+            # create an in-memory database that's non-persistent
+            self._con = duckdb.connect(":memory:")
+        else:
+            # connect to motherduck
+            md_token = current_context().secrets.get(
+                group=md_secret.group,
+                key=md_secret.key,
+                group_version=md_secret.group_version,
+            )
+            self._con = duckdb.connect("md:", config={'motherduck_token': md_token})
```

Comment:
What is the recommended way to connect to motherduck? In their docs, it uses `md:?motherduck_token=<token>`.

Reply:
They recommend both ways. I thought this was was cleaner and more readable.